bi_sql_editor/models/bi_sql_view_field.py

```diff
@@ -142,41 +145,42 @@ def _compute_index_name(self):
             )
 
     # Overload Section
-    @api.model
-    def create(self, vals):
-        field_without_prefix = vals["name"][2:]
-        # guess field description
-        field_description = re.sub(
-            r"\w+",
-            lambda m: m.group(0).capitalize(),
-            field_without_prefix.replace("_id", "").replace("_", " "),
-        )
-
-        # Guess ttype
-        # Don't execute as simple .get() in the dict to manage
-        # correctly the type 'character varying(x)'
-        ttype = False
-        for k, v in self._SQL_MAPPING.items():
-            if k in vals["sql_type"]:
-                ttype = v
-
-        # Guess many2one_model_id
-        many2one_model_id = False
-        if vals["sql_type"] == "integer" and (vals["name"][-3:] == "_id"):
-            ttype = "many2one"
-            model_name = self._model_mapping().get(field_without_prefix, "")
-            many2one_model_id = (
-                self.env["ir.model"].search([("model", "=", model_name)]).id
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            field_without_prefix = vals["name"][2:]
+            # guess field description
+            field_description = re.sub(
+                r"\w+",
+                lambda m: m.group(0).capitalize(),
+                field_without_prefix.replace("_id", "").replace("_", " "),
             )
 
-        vals.update(
-            {
-                "ttype": ttype,
-                "field_description": field_description,
-                "many2one_model_id": many2one_model_id,
-            }
-        )
-        return super(BiSQLViewField, self).create(vals)
+            # Guess ttype
+            # Don't execute as simple .get() in the dict to manage
+            # correctly the type 'character varying(x)'
+            ttype = False
+            for k, v in self._SQL_MAPPING.items():
+                if k in vals["sql_type"]:
+                    ttype = v
+
+            # Guess many2one_model_id
+            many2one_model_id = False
+            if vals["sql_type"] == "integer" and (vals["name"][-3:] == "_id"):
+                ttype = "many2one"
+                model_name = self._model_mapping().get(field_without_prefix, "")
+                many2one_model_id = (
+                    self.env["ir.model"].search([("model", "=", model_name)]).id
+                )
+
+            vals.update(
+                {
+                    "ttype": ttype,
+                    "field_description": field_description,
+                    "many2one_model_id": many2one_model_id,
+                }
+            )
+        return super().create(vals)
```

Comment:
It seems quite wrong to me, should be `vals_list` instead of `vals` right ?

Reply:
Good catch

Reply:
thanks for your review ! 
fixed.